<?xml version="1.0"?>

<launch>

    <arg name="max_distance" default="20.0"/>
    <arg name="labels_path" default="$(find semantic_segmentation_with_point_cloud_integrator)/config/cityscapes.json"/>
    <arg name="extraction_classes" default="ground, road, sidewalk, terrain"/>

    <node pkg="semantic_segmentation_with_point_cloud_integrator" type="semantic_segmentation_with_point_cloud_integrator" name="semantic_segmentation_with_point_cloud_integrator">
        <param name="MAX_DISTANCE" value="$(arg max_distance)"/>
        <param name="LABELS_PATH" value="$(arg labels_path)"/>
        <param name="EXTRACTION_CLASSES" value="$(arg extraction_classes)"/>
		<remap from="/deeplab/segmented_image" to="/deeplab/segmented_image"/>
        <remap from="/camera_info" to="/camera_info"/>
        <remap from="/pointcloud" to="/velodyne_points"/>
        <remap from="/projection/raw" to="/projection/raw"/>
        <remap from="/projection/semantic" to="/projection/semantic"/>
        <remap from="/cloud/colored/raw" to="/cloud/colored/raw"/>
        <remap from="/cloud/colored/semantic" to="/cloud/colored/semantic"/>
    </node>

</launch>
